src/toast/types.ts

```diff
@@ -0,0 +1,21 @@
+export enum TOAST_COLORS {
+  SUCCESS = "success",
+  ERROR = "error",
+  INFO = "info",
+  INVERSE = "inverse",
+  WARNING = "warning",
+}
+export type ToastOptions = {
+  color?: TOAST_COLORS;
+  showDismissButton?: boolean;
+  dismissOnClickingOutside?: boolean;
+  autoHideDuration?: number;
+  onClose?: () => void;
+  onOpen?: () => void;
+}
+
+type ToastProps = ToastOptions & {
+  message: string;
+};
+
```

Comment:
types are redundant. they will be inferred using declare module statement will a

```ts
declare global {
  interface HTMLElementTagNameMap {
    "tap-toast": TapToast;
  }
}
```